server/src/interfaces/notification.interface.ts

```diff
@@ -0,0 +1,51 @@
+export const INotificationRepository = 'INotificationRepository';
+
+export type SendEmailOptions = {
+  from: string;
+  to: string;
+  replyTo?: string;
+  subject: string;
+  html: string;
+  text: string;
+  smtp: SmtpOptions;
+};
+
+export type SmtpOptions = {
+  host: string;
+  port?: number;
+  username?: string;
+  password?: string;
+  ignoreCert?: boolean;
+};
+
+export enum EmailTemplate {
+  WELCOME = 'welcome',
+  RESET_PASSWORD = 'reset-password',
+}
+
+export interface WelcomeEmailProps {
+  baseUrl: string;
+  displayName: string;
+  username: string;
+  password?: string;
+}
+
+export interface ResetPasswordEmailProps {
+  user: string;
+  email: string;
```

Comment:
In line with the other comment about the reset password template, this should likely be removed, but if it isn't surely we will need another property for the link that would allow them to actually reset their password? Probably still worth just removing for now.

Reply:
Removed this.